### ISSUE TITLE
"Print Test Page" in GNOME results in error

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#17990]
## Issue description
Setting up a printer in GNOME and pressing "Print Test Page" (in system-config-printer) results in this error dialog:

```
CUPS server error
There was an error during the CUPS operation: 'No such file or directory'.
```

AFAIK, this is not a regression, it has simply never worked.

### Steps to reproduce
```
$ nix-build -A system-config-printer
$ ./result/bin/system-config-printer
... add printer
... select "Print Test Page"
... watch the error message appear and no print being done
```

## Technical details

* System: NixOS 16.09.git.6bfe459 (Flounder) (latest master as of today)
* Nix version: 1.11.2
* Nixpkgs version: b71d71c576df3dbac8af76241260246010e08536  (latest master as of today)

**Comment by bjornfor** [comment]
I increased cups debug level with ```sudo cupsctl LogLevel=debug2```. Didn't spot the problem.

Started system-config-printer again, now with ```--debug``` flag:
* delete printer
* add printer
* select "Print Test Page" results in
```
...
Printing default test page
Authentication pass: 1
Authentication: password callback set
1030: 'No such file or directory'
```

**Comment by bjornfor** [comment]
After adding a small patch to system-config-printer I get a tiny bit more detail: ```cups.IPPError: (1030, 'No such file or directory')```

**Comment by bjornfor** [comment]
Running ```system-config-printer``` under strace found it:

```
2033  access("/usr/share/cups/data/testprint", R_OK) = -1 ENOENT (No such file or directory)
2033  access("/usr/share/cups/data/testprint.ps", R_OK) = -1 ENOENT (No such file or directory)
2033  access("/usr/local/share/cups/data/testprint", R_OK) = -1 ENOENT (No such file or directory)
2033  access("/usr/local/share/cups/data/testprint.ps", R_OK) = -1 ENOENT (No such file or directory)
2033  open("/usr/share/cups/data/testprint", O_RDONLY) = -1 ENOENT (No such file or directory)
```

**Comment by bjornfor** [comment]
Ugh, system-config-printer has lots of /usr/share, /usr/lib hardcoded strings...

**Comment by bjornfor** [comment]
So the ```testprint``` file it's looking for comes from the ```cups-filters``` package. Should we make it compile time or runtime dependency? cc @abbradar (has done lots of printing stuff lately, AFAIR).

**Comment by matthewbauer** [comment]
It looks like the offending part is here:

https://github.com/Distrotech/pycups/blob/140c5591190b7dfc2516eee7ef9cc0769696ce4c/cupsconnection.c#L3965-L3967

Perhaps we can set CUPS_DATADIR, somewhere?

**Comment by abbradar** [comment]
Try wrapping `system-config-printer` with `CUPS_DATADIR=${cups-filters}/share/cups` (as @matthewbauer suggested but give it `cups-filters` instead) -- it may help.

**Comment by bjornfor** [comment]
@matthewbauer: Ah, there it is! I was looking in the wrong places (cups), found some false positives. I'll make a patch.

**Comment by bjornfor** [comment]
@abbradar: Will do.

**Comment by bjornfor** [comment]
Fixed in 90de10b3eeb5b327d1aa1c70564437fe1f3021fb.
